IdentityCore/src/requests/sdk/MSIDTokenResponseValidator.m

```diff
@@ -240,6 +243,21 @@ - (MSIDTokenResult *)validateAndSaveTokenResponse:(MSIDTokenResponse *)tokenResp
     {
         return nil;
     }
+    
+    //save metadata
+    NSError *updateMetadataError = nil;
+    [metadataCache updateAuthorityURL:tokenResult.authority.url
```

Comment:
should this be done only for MSAL?

Reply:
Let's have it nullable and have ADAL pass in nil.